README.md

```diff
@@ -456,6 +459,129 @@ remote_state = {
 parent, but the `remote_state` settings of the child. 
 
+### The spin-up and tear-down commands
+
+Let's say you have a single environment (e.g. `stage` or `prod`) that has a number of Terraform modules within it:
+
+```
+my-terraform-repo
+  └ .terragrunt
+  └ stage
+      └ frontend-app
+          └ main.tf
+          └ .terragrunt
+      └ backend-app
+          └ main.tf
+          └ .terragrunt
+      └ search-app
+          └ main.tf
+          └ .terragrunt
+      └ mysql
+          └ main.tf
+          └ .terragrunt
+      └ redis
+          └ main.tf
+          └ .terragrunt
+      └ vpc
+          └ main.tf
+          └ .terragrunt
+```
+
+There is one module to deploy a frontend-app, another to deploy a backend-app, another for the MySQL database, and so 
+on. To deploy such an environment, you'd have to manually run `terragrunt apply` in each of the subfolders. How do you
+avoid this tedious and time-consuming process?
+
+The answer is that you can use the `spin-up` command:
+ 
+```
+cd my-terraform-repo/stage
+terragrunt spin-up
+```
+
+When you run this command, Terragrunt will find all `.terragrunt` files in the subfolders of the current working 
+directory, and run `terragrunt apply` in each one concurrently.  
+
+Similarly, to undeploy all the Terraform modules, you can use the `tear-down` command:
+
+```
+cd my-terraform-repo/stage
+terragrunt tear-down
+```
+
+Of course, if your modules have dependencies between them—for example, you can't deploy the backend-app until the MySQL
+database is deployed—you'll need to express those dependencines in your `.terragrunt` config as explained in the next
+section.
+
+### Dependencies between modules
+
+Consider the following file structure for the `stage` environment:
+
+```
+my-terraform-repo
+  └ .terragrunt
+  └ stage
+      └ frontend-app
+          └ main.tf
+          └ .terragrunt
+      └ backend-app
+          └ main.tf
+          └ .terragrunt
+      └ search-app
+          └ main.tf
+          └ .terragrunt
+      └ mysql
+          └ main.tf
+          └ .terragrunt
+      └ redis
+          └ main.tf
+          └ .terragrunt
+      └ vpc
+          └ main.tf
+          └ .terragrunt
+```
+
+Let's assume you have the following dependencies between Terraform modules:
+
+* Every module depends on the VPC being deployed
+* The backend-app depends on the MySQL database and Redis
+* The frontend-app and search-app depend on the backend-app
+
+You can express these dependencies in your `.terragrunt` config files using the `dependencies` block. For example, in
+`stage/backend-app/.terragrunt` you would specify:
+
+```hcl
+include = {
+  path = "${find_in_parent_folders()}"
+}
+
+dependencies = {
+  paths = ["../vpc", "../mysql", "../redis"]
+}
+```
+
+Similarly, in `stage/frontend-app/.terragrunt`, you would specify:
+
+```hcl
+include = {
+  path = "${find_in_parent_folders()}"
+}
+
+dependencies = {
+  paths = ["../vpc", "../backend-app"]
+}
+```
+
```

Comment:
Did we ultimately rule out directly parsing `terraform_remote_state` because it required that we resolve the Terraform template in order to discover the dependency?